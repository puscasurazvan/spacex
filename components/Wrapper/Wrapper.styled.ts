import styled from 'styled-components';

export const Container = styled.div({
  display: 'flex',
  flexWrap: 'wrap',
  padding: '20px',
  lineHeight: '20px',
});
