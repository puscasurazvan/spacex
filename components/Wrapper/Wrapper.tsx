import { ReactNode } from 'react';

import * as Styled from './Wrapper.styled';

type WrapperProps = {
  children: ReactNode;
};

export const Wrapper = ({ children }: WrapperProps) => {
  return <Styled.Container data-testid="card-wrapper">{children}</Styled.Container>;
};
