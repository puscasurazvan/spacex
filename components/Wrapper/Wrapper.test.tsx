import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Wrapper } from './Wrapper';

describe('Wrapper Component', () => {
  it('renders the Wrapper component with children', () => {
    const children = <div>Test Children</div>;

    const { container } = render(<Wrapper>{children}</Wrapper>);

    expect(container).toMatchSnapshot();
    expect(screen.getByTestId(`card-wrapper`)).toBeInTheDocument();
  });
});
