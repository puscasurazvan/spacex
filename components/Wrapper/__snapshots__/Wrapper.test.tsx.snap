// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper Component renders the Wrapper component with children 1`] = `
<div>
  <div
    class="sc-bczSft OKzct"
    data-testid="card-wrapper"
  >
    <div>
      Test Children
    </div>
  </div>
</div>
`;
